src/resources/assets/assetsApi.ts

```diff
@@ -90,39 +87,127 @@ export class AssetsAPI {
    */
   public delete: AssetDeleteEndpoint;
 
+  private client: CogniteClient;
+
   /** @hidden */
-  constructor(project: string, instance: AxiosInstance, map: MetadataMap) {
+  constructor(
+    client: CogniteClient,
+    project: string,
+    instance: AxiosInstance,
+    map: MetadataMap
+  ) {
+    this.client = client;
     const path = projectUrl(project) + '/assets';
-    this.create = generateCreateEndpoint(instance, path, map, assetChunker);
-    this.list = generateListEndpoint(instance, path, map, true);
-    this.retrieve = generateRetrieveEndpoint(instance, path, map);
-    this.update = generateUpdateEndpoint(instance, path, map);
-    this.search = generateSearchEndpoint(instance, path, map);
+    const transformResponse = this.transformToAssetListObject;
+    this.create = generateCreateEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse,
+      assetChunker
+    );
+    this.list = generateListEndpoint(
+      instance,
+      path,
+      map,
+      true,
+      transformResponse
+    );
+    this.retrieve = generateRetrieveEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.update = generateUpdateEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.search = generateSearchEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
     this.delete = generateDeleteEndpointWithParams(instance, path, map);
   }
+
+  public async retrieveSubtree(id: types.CogniteInternalId, depth: number) {
+    const currentDepth: number = 0;
+    const rootAssetList = await this.retrieve([{ id }]);
+    return this.getAssetSubtree(rootAssetList, currentDepth, depth);
```

Comment:
maybe you don't need `currentDepth ` variable? just pass 0?

Reply:
Isn't this way more explainable? Then you don't have to check `getAssetSubtree` to understand what the 0 is.

Reply:
agree